kratos/utilities/time_discretization.h

```diff
@@ -0,0 +1,113 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+#if !defined(KRATOS_TIME_DISCRETIZATION_H_INCLUDED )
+#define  KRATOS_TIME_DISCRETIZATION_H_INCLUDED
+
+// System includes
+#include <vector>
+
+// External includes
+
+// Project includes
+
+namespace Kratos {
+namespace TimeDiscretization {
+
+class BDF1
+{
+public:
+    void ComputeBDFCoefficients(const double DeltaTime,
+                                std::vector<double>& rCoefficients) const;
+};
+
+class BDF2
+{
+public:
+    void ComputeBDFCoefficients(const double DeltaTime,
+                                std::vector<double>& rCoefficients) const;
+};
+
+class BDF3
+{
+public:
+    void ComputeBDFCoefficients(const double DeltaTime,
```

Comment:
here you need to pass the ProcessInfo, since you will need the current time and the old time(s)

Reply:
look at how I solved it now, I am quite convinced this is a better solution